docs/rules/no-unused-expressions.md

```diff
@@ -39,6 +39,18 @@ a && b()
 a, b()
 
 c = a, b;
+
+(function namedFunctionInExpressionContext () {f();});
```

Comment:
You might be interested to think of clear contrasts in example code for an ESLint rule as somewhat similar to the following concepts in descriptive linguistics to give convincing evidence whether two sounds are phonemes or allophones: http://clas.mq.edu.au/speech/phonetics/phonology/phoneme/
* contrast in identical environment, also known as minimal pair: https://en.wikipedia.org/wiki/Minimal_pair
* contrast in analogous environment